### PR TITLE
tests/operator, instancetypes, preferences: Replace Update with Patch

### PR DESCRIPTION
**Pull request by RamLavi**
### What this PR does
It is common practice on our e2e tests to use Update k8s API in order to run the test. 
The issue is that most often tests do not really need to use Update (for an atomic change) and Patch would be quicker and more precises.
This PR is replacing an unneeded use of update in favor of patch in the "Should revert changes" context.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/operator/operator.go [hunk 1]
```diff
@@ -2663,16 +2663,14 @@ spec:
...
-				}
+				patchDataBytes, err := patch.New(
+					patch.WithAdd(fmt.Sprintf("/metadata/annotations/%s", keyTest), valModified),
+					patch.WithAdd(fmt.Sprintf("/metadata/labels/%s", keyTest), valModified),
+					patch.WithReplace("/spec/cpu", v1beta1.CPUInstancetype{Guest: cpu}),
+				).GeneratePayload()
+				Expect(err).ToNot(HaveOccurred())
 
-				instancetype, err = virtClient.VirtualMachineClusterInstancetype().Update(context.Background(), instancetype, metav1.UpdateOptions{})
+				instancetype, err := virtClient.VirtualMachineClusterInstancetype().Patch(context.Background(), originalInstancetype.Name, types.JSONPatchType, patchDataBytes, metav1.PatchOptions{})
```
Comment (akalenyu) [inline tests/operator/operator.go:2673]: So while PATCH is surely a better, more precise operation I think using it here may hide conflicts that should not be occurring.
IIUC, the test enables the feature, instancetypes get deployed, and from that point on they should not be changing at all.

**Review by akalenyu — COMMENTED** [review]
We could get this in, but I think update is just fine here, unless we've seen this fail
(which could mean there's a bug!)

File: tests/operator/operator.go (hunk 1 above)
Comment (RamLavi) [reply tests/operator/operator.go:2673]: Not sure I agree here. Here's why: If we use Patch and there's a "conflict" (not as in "returns a conflict error" which will not happen now, but rather as "someone else did another change at the same time and his change won"), then won't the `Except` right after the Patch occasionally fail?
IMO You get the same end result, only you're using a more proper tool.

Comment (akalenyu) [reply tests/operator/operator.go:2673]: > then won't the `Except` right after the Patch occasionally fail?

It won't fail if that other entity changed different fields on the CR.
In that case, everything will still go through, and we'll never know that the instacetype CR is hotlooping over some field

Comment (RamLavi) [reply tests/operator/operator.go:2673]: So you want to keep the update in order to *make sure* no other entity modifies it as well?
Are you sure that's the best way to do that?
How can you ensure that if it fails on conflict in the future, that the person trying to fix it won't just put retryOnConflict?

**Comment by RamLavi** [comment]
Closing as there is no consensus this is a good case to move to patch.

**Pull request closed without merging**